adserver/models.py

```diff
@@ -1073,6 +1082,17 @@ def incr(self, impression_type, publisher, div_id=None, ad_type_slug=None):
                 **{impression_type: models.F(impression_type) + 1}
             )
 
+        if publisher.record_geos and request:
```

Comment:
Taken by itself, I'm not too concerned here. However, the long term solution is to rollup this data every few hours or every day. I'd like to do as little as possible in the request to keep the performance fast and to keep it simple. I do wonder whether this should just wait for the infrastructure changes needed to do that. I feel like we're writing code we're going to throw away (models wouldn't go away but this section would).